File: sample/app/src/main/java/com/anggrayudi/hiddenapi/sample/Adapter.java
package com.anggrayudi.hiddenapi.sample;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by Anggrayudi on 11/03/2016.
 */
class Adapter extends BaseAdapter {

    private static class ViewHolder {
        TextView source, result, desc;
    }

    private ArrayList<Model> models;

    Adapter(ArrayList<Model> models){
        this.models = models;
    }

    @Override
    public int getCount() {
        return models.size();
    }

    @Override
    public Object getItem(int position) {
        return models.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater)
                    parent.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.list_layout, parent, false);

            ViewHolder viewHolder = new ViewHolder();
            viewHolder.source = convertView.findViewById(R.id.source);
            viewHolder.result = convertView.findViewById(R.id.result);
            viewHolder.desc = convertView.findViewById(R.id.desc);
            convertView.setTag(viewHolder);
        }

        ViewHolder holder = (ViewHolder) convertView.getTag();
        Model model = models.get(position);

        holder.source.setText(model.source);

        holder.result.setVisibility(model.result.equals("Result = ") ? View.GONE : View.VISIBLE);
        holder.result.setText(model.result);

        holder.desc.setVisibility(TextUtils.isEmpty(model.description) ? View.GONE : View.VISIBLE);
        holder.desc.setText(model.description);

        return convertView;
    }
}

File: sample/src/main/java/com/anggrayudi/hiddenapi/sample/Adapter.kt
package com.anggrayudi.hiddenapi.sample

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

/**
 * Created by Anggrayudi on 11/03/2016.
 */
class Adapter(private val models: List<Model> = emptyList()) : BaseAdapter() {

    private class ViewHolder(
        val source: TextView,
        val result: TextView,
        val desc: TextView
    )

    override fun getCount() = models.size

    override fun getItem(position: Int) = models[position]

    override fun getItemId(position: Int) = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            val inflater = parent.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.list_layout, parent, false)
            view.tag = ViewHolder(
                view.findViewById(R.id.source),
                view.findViewById(R.id.result),
                view.findViewById(R.id.desc)
            )
        }
        val holder = view!!.tag as ViewHolder
        val (source, result, description) = models[position]
        holder.source.text = source
        holder.result.visibility = if (result.isNullOrEmpty()) View.GONE else View.VISIBLE
        holder.result.text = "Result = $result"
        holder.desc.visibility = if (description.isNullOrEmpty()) View.GONE else View.VISIBLE
        holder.desc.text = description
        return view
    }
}